server/cluster/cluster_worker.go

```diff
@@ -41,6 +41,49 @@ func (c *RaftCluster) HandleRegionHeartbeat(region *core.RegionInfo) error {
 	return nil
 }
 
+// ProcessRegionSplit to process split region into region cache.
+// it's different with the region heartbeat, it's only fill some new region into the region cache.
+// so it doesn't consider the leader and hot statistics.
+func (c *RaftCluster) ProcessRegionSplit(regions []*metapb.Region) []error {
+	if err := c.checkSplitRegions(regions); err != nil {
+		return []error{err}
+	}
+	total := len(regions) - 1
+	regions[0], regions[total] = regions[total], regions[0]
+	leaderStoreID := uint64(0)
+	if r := c.core.GetRegion(regions[0].GetId()); r != nil {
+		leaderStoreID = r.GetLeader().GetStoreId()
+	}
+	if leaderStoreID == 0 {
+		return []error{errors.New("origin region no leader")}
+	}
+	errList := make([]error, 0, total)
+	for _, region := range regions {
+		if len(region.GetPeers()) == 0 {
+			errList = append(errList, errors.New(fmt.Sprintf("region:%d has no peer", region.GetId())))
+			continue
+		}
+		// region split initiator store will be leader with a high probability
+		leader := region.Peers[0]
```

Comment:
Will it cause the balance leader to wrong place?

Reply:
good catch.

Reply:
I will mark this store as `split store` and balance region scheduler doesn't pick this store as target store.